frontend/features/plugins/useSitePlugin.ts

```diff
@@ -30,23 +46,63 @@ export interface PageDataCreateInput {
   locale: string;
 }
 
-export function usePagePlugin(pageData: PageData): [PageData, Form] {
+export interface GlobalData {
+  id: string;
+  topbar: {
+    id: string;
+    menu: MenuData;
+  };
+}
+
+export interface MenuData {
+  id: string;
+  links: NavBlockData[];
+}
+
+export interface AllData {
+  global: {
+    id: string;
+    topbar: {
+      id: string;
+      menu: {
+        id: string;
+        links: NavBlockData[];
+      };
+    };
+  };
+  page: {
+    id: string;
+    title?: string;
+    path?: string;
+    sections: SectionBlockData[];
+  };
+}
+
+export function usePagePlugin(allData: AllData): [AllData, Form] {
   const cms = useCMS();
   const router = useRouter();
-  const formConfig: FormOptions<PageData> = {
-    id: pageData,
-    label: "Page",
-    initialValues: pageData,
-    onSubmit: async (values) => {
-      const input = getPageInput(values);
+
+  const formConfig: FormOptions<AllData> = {
+    id: 1,
+    label: "All",
```

Comment:
What's this?

Reply:
"id" is the unique identifier of the form, I've set this to 1 because we have just one form.
"label" is just a form name that will appear in the sidebar. It's required

Reply:
The Tina documentations says:

<img width="759" alt="Screen Shot 2021-07-28 at 10 19 00" src="https://user-images.githubusercontent.com/4640135/127288954-865f5a5f-ecbf-4fe1-b202-1a68f6a5a1d0.png">

> This should be derived from the content 

Doesn't make sense for to different pages to have the same form id right? If it's not the same piece of data, it shouldn't have the same id.. When you are trying to hard code an id red flags should rise in your head.
Why not just using the page id here?

The label "All" doesn't mean much, it's way too vague.. Maybe "Page settings" ?

Reply:
👍🏻